crates/matrix-sdk-base/src/rooms/normal.rs

```diff
@@ -621,6 +653,12 @@ impl RoomInfo {
         self.base_info.encryption.is_some()
     }
 
+    /// Set the encryption event content in this room.
+    pub fn set_encryption_event(&mut self, event: Option<RoomEncryptionEventContent>) -> &Self {
```

Comment:
Why does this return `&Self`?

Reply:
Oh, that would actually need to be `&mut Self` right. If that is still a question with mut self: just to allow to chain setters, it is a general pattern

Reply:
It doesn't seem like the other streets here are chainable, so I don't think it makes a lot of sense to make this one chainable.

Reply:
Alright